instrumentation/log4j/log4j-context-data/log4j-context-data-2.17/library-autoconfigure/src/main/java/io/opentelemetry/instrumentation/log4j/contextdata/v2_17/OpenTelemetryContextDataProvider.java

```diff
@@ -67,6 +68,11 @@ public Map<String, String> supplyContextData() {
       return staticContextData;
     }
 
+    if (ThreadContext.containsKey(ContextDataKeys.TRACE_ID_KEY)) {
+      // Assume already instrumented event if traceId is present.
+      return staticContextData;
+    }
```

Comment:
what is the use case where the user's trace id would be different from the opentelemetry trace id?

Reply:
Note that there is a similar check in logback https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/d45ebfa7773e06d42d0710c7799ae776560e0c45/instrumentation/logback/logback-mdc-1.0/library/src/main/java/io/opentelemetry/instrumentation/logback/mdc/v1_0/OpenTelemetryAppender.java#L79 and log4j 1.2 https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/d45ebfa7773e06d42d0710c7799ae776560e0c45/instrumentation/log4j/log4j-mdc-1.2/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/log4j/mdc/v1_2/LoggingEventInstrumentation.java#L56

Reply:
thanks 👍